src/common/s3_adapter.cpp

```diff
@@ -340,13 +343,18 @@ void S3Adapter::PutObjectAsync(std::shared_ptr<PutObjectAsyncContext> context) {
         AWS_ALLOCATE_TAG, context->buffer, context->bufferSize));
 
     auto originCallback = context->cb;
-    auto wrapperCallback =
+    auto throttledCallback =
         [this,
-         originCallback](const std::shared_ptr<PutObjectAsyncContext>& ctx) {
+         originCallback](const std::shared_ptr<PutObjectAsyncContext> ctx) {
             inflightBytesThrottle_->OnComplete(ctx->bufferSize);
             ctx->cb = originCallback;
             ctx->cb(ctx);
         };
+    auto asyncCallback =
+        [this,
+         throttledCallback](const std::shared_ptr<PutObjectAsyncContext>& ctx) {
+            asyncCallbackTP_.Enqueue(std::move(throttledCallback), ctx);
```

Comment:
can we just do the Enqueue in L378 rather than wrap another call back?

Reply:
yes, i'll put it in handle function